mudslide/turbomole_model.py

```diff
@@ -0,0 +1,192 @@
+            c_list = c.split()
+            self.atom_order.append(c_list[3])
+            coords.append([float(val) for val in c_list[:3]])
+            self.mass.append(3 * [masses[c_list[3]]])
+
+        self.ndim_ = 3 * len(coords)
+        self.X = np.array(coords, dtype=np.float64).reshape(self.ndim())
+        self.mass = np.array(self.mass, dtype=np.float64).reshape(self.ndim()) * amu_to_au
+
+
+    def update_coords(self, X):
+        X = X.reshape((self.ndim() // 3, 3))
+
+        with open(self.coord_path, "r") as f:
+            lines = f.readlines()
+
+        regex = re.compile(r"\$coord\n|\$coord\s")
+        coordline = 0
+        for i, line in enumerate(lines):
+            if regex.match(line) is not None:
+                coordline = i
+                break
+        # Reached end of file without finding $coord.
+        if line == "":
+            raise ValueError(f"$coord entry not found in file: {self.coord_path}!")
+
+        coordline +=1
+        for i, coord_list in enumerate(X):
+            lines[coordline] = "{:20.14f}{:22.14f}{:22.14f}{:>7}\n".format( 
+                                  coord_list[0], coord_list[1], coord_list[2], self.atom_order[i]
+                    )
+            coordline += 1
+
+        with open(self.coord_path, "w") as coord_file:
+            coord_file.write("".join(lines))            
+
+        # Now add results to model
+    def call_turbomole(self, outname="turbo.out"):
+
+        with open(outname, "w") as f:
+            for turbomole_module in self.turbomole_modules.values():
+                tur_output = subprocess.run(turbomole_module, stdout=f) 
+
+        # Parse results with Turboparse
+        with open(outname, "r") as f:
+            data_dict = turboparse.parse_turbo(f)
+
+        # Now add results to model
+        energy = data_dict[self.turbomole_modules["gs_energy"]]["energy"]
+        self.energies = [energy]
+        parsed_gradients = data_dict[self.turbomole_modules["gs_grads"]]["gradient"]
+
+        # Check for presence of egrad turbomole module
+
+        if "egrad" in self.turbomole_modules.values():
+            # egrad updates to energy
+            parsed_energies = data_dict["egrad"]["excited_state"][0]["energy"]
+            excited_energies = [
+                data_dict["egrad"]["excited_state"][i]["energy"]
+                + energy for i in range(len(data_dict["egrad"]["excited_state"]))
+            ]
+            self.energies.extend(excited_energies)
+
+            # egrad couplings
+            parsed_nac_coupling = data_dict["egrad"]["coupling"]
+            self.derivative_coupling = np.zeros((self.nstates(), self.nstates(), self.ndim()))
+            for dct in parsed_nac_coupling:
+                i = dct["bra_state"]
+                j = dct["ket_state"]
+
+                self.derivative_coupling[i][j] = np.array(dct["d/dR"]).reshape(self.ndim(), order="F")
+                self.derivative_coupling[j][i] = self.derivative_coupling[i][j]
```

Comment:
This line is wrong. Derivative couplings are anti-Hermitian, not Hermitian. Meaning tau_ij = -tau_ji.

Reply:
should have been checking this  - corrected now! 

<img width="1034" alt="Screen Shot 2022-04-19 at 2 06 38 PM" src="https://user-images.githubusercontent.com/46786940/164067915-80eb2f81-f708-485c-b830-a652d452b126.png">